### PR TITLE
Remove pip dependencies

### PR DESCRIPTION
**Pull request by fievelk**
Adding `pip` and `pip-tools` to the requirements is considered to be unsafe and it's generating errors/warnings because of incompatibilities between some of our dependencies. Updating `pip-tools` should be manually done by the developer before installing the requirements.

- remove `pip` and `pip-tools` from explicit requirements
- re-compile requirements (using `pip-compile`)

### CODE REVIEW

File: integration_tests/build_test_containers.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@
 set -e
 
 D3A_IMAGE_TAG="d3a-tests"
-REDIS_IMAGE_NAME="gsyd3a/d3a:redis-staging"
+REDIS_IMAGE_NAME="redis:6.2.5"
 
 if [[ "$(docker images -q ${REDIS_IMAGE_NAME} 2> /dev/null)" == "" ]]; then
     docker login -u "${DOCKER_USERNAME}" -p "${DOCKER_PASSWORD}" &&
```
Comment (hannesdiedrich) [inline integration_tests/build_test_containers.sh:9]: You can delete all this, as `docker run` will pull automatically. Thanks!

**Review by hannesdiedrich — APPROVED** [review]
One comment, once resolved, LGTM

File: integration_tests/build_test_containers.sh (hunk 1 above)
Comment (fievelk) [reply integration_tests/build_test_containers.sh:9]: @hannesdiedrich will `docker run` also execute the login?

Comment (hannesdiedrich) [reply integration_tests/build_test_containers.sh:9]: No need to login any more, as redis is publicly available on dockerhub.

**Review by hannesdiedrich — APPROVED** [review]
LGTM

**Review by faizan2590 — APPROVED** [review]
LGTM

**Pull request merged**